src/main/java/com/coreos/jetcd/LeaseImpl.java

```diff
@@ -238,11 +245,27 @@ public void onCompleted() {
 
     // cancel grpc stream when leaseKeepAliveResponseCompletableFuture completes.
     // TODO: Have a internal executor to manage Async threads?
-    lkaFuture.whenCompleteAsync((val, throwable) -> requestObserver.onCompleted());
+    lkaFuture
+        .whenCompleteAsync((val, throwable) -> requestObserver.onCompleted(), this.executorService);
 
     return lkaFuture;
   }
 
+  @Override
+  public CompletableFuture<com.coreos.jetcd.lease.LeaseTimeToLiveResponse> timeToLive(long leaseId,
+      LeaseOption option) {
+    checkNotNull(option, "LeaseOption should not be null");
```

Comment:
I saw the kvImpl doesn't allow `null`l option. so I follow that. I wonder which one seems more correct?
`timeToLive(leaseID, LeaseOption.DEFAULT)` or`timeToLive(leaseID, null)`